fastlane/lib/fastlane/actions/oclint.rb

```diff
@@ -23,7 +23,11 @@ def self.run(params)
         select_regex = params[:select_regex] if params[:select_regex] # Overwrite deprecated select_reqex
         exclude_regex = params[:exclude_regex]
 
-        files = JSON.parse(File.read(compile_commands)).map { |compile_command| compile_command['file'] }
+        files = JSON.parse(File.read(compile_commands)).map do |compile_command|
+          file = compile_command['file']
+          File.exist?(file) ? file : compile_command['directory'] + '/' + file
```

Comment:
From your example data, I think you would get a path like:

/Users/radbuilder/Documents/Test/App_Apps_ios/staticlib/**StaticLibShared/StaticLibShared**/ClassStaticLibShared.m

Is that doubling of the **StaticLibShared** part right?